### PR TITLE
[PLAY-194] Contact Kit Update: Extension Variant

### PR DESCRIPTION
**Pull request by augustomallmann**
React Kit Coverage
#### Screens

[INSERT SCREENSHOT]

Screen Shot 2022-07-12 at 2 33 54 PM

#### Breaking Changes

No breaking changed.  Added a new `extention` option to the prop `contactType`, making it possible to display 4-digit phone extensions.

#### Runway Ticket URL

[PLAY-194](https://nitro.powerhrg.com/runway/backlog_items/PLAY-194)

#### How to test this

[INSERT TESTING DETAILS]

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [ ] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_contact/contact.rb [hunk 1]
```diff
@@ -35,8 +37,12 @@ def contact_icon
       def formatted_contact_value
         if contact_type == "email"
           contact_value
+        elsif contact_type == "extension" && contact_value.match(/^\d{4}$/)
+          contact_value
+        elsif contact_type != "email" && contact_type != "extension"
+          number_to_phone(formatted_value, area_code: true, raise: true)
         else
-          number_to_phone(formatted_value, area_code: true)
+          ""
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_contact/contact.rb:45]: This area makes me nervous. Before, we had an else that would call `number_to_phone` now we don't. Can you explain?

Comment (augustomallmann) [reply playbook/app/pb_kits/playbook/pb_contact/contact.rb:45]: You're right, reviewing it makes me realize that it'd not get to the last `else` cuz everything that's not an `email` nor `extension` will be resolved on the condition before. My idea was to return an empty string when a `contact_type` wasn't specified. My suggestion is to return to the original code as the example below, what do you think?

```
if contact_type == "email"
  contact_value
elsif contact_type == "extension" && contact_value.match(/^\d{4}$/)
  contact_value
else
  number_to_phone(formatted_value, area_code: true, raise: true)
end
```

Comment (jasperfurniss) [reply playbook/app/pb_kits/playbook/pb_contact/contact.rb:45]: I think the way you have it written in your comment from 5 days ago is the safest. 
Go ahead and adjust to that, and i'll give this a final review.

**Comment by jasperfurniss** [comment]
@augustomallmann Did you happen to test the `raise: true` that you implemented? 
```
If true, raises [InvalidNumberError](https://apidock.com/rails/ActionView/Helpers/NumberHelper/InvalidNumberError) when the argument is invalid.
```

I'm just curious what happens UI-wise, if someone has an invalid number.

**Comment by augustomallmann** [comment]
> @augustomallmann Did you happen to test the `raise: true` that you implemented?
> 
> ```
> If true, raises InvalidNumberError
> when the argument is invalid.
> ```
> 
> I'm just curious what happens UI-wise, if someone has an invalid number.

Hey @jasperfurniss. I was testing the `raise` parameter only to check what would happen if I pass a string instead a number to that function. I removed it from the code now since it wasn't suppose to be there. Thanks !!

**Review by jasperfurniss — APPROVED** [review]

**Pull request merged**